File: classify-android/app/src/main/java/io/classify/service/UserService.java
package io.classify.service;

import com.remswork.project.alice.model.UserDetail;

import io.classify.model.User;
import io.reactivex.Observable;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.PUT;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface UserService {

    @GET("userDetail")
    Observable<User> getByUsername(@Query("username") String username);

    @PUT("userDetail/{userId}")
    Observable<User> update(@Path("userId") long userId, @Body UserDetail userDetail);
}

File: classify-android-integration/app/src/main/java/io/classify/data/remote/service/UserService.kt
package io.classify.data.remote.service

import io.classify.data.model.User
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface UserService {

    @GET("users/{id}")
    fun getById(@Path("id") id: Long): Observable<User>

    @GET("users")
    fun getByUsername(@Query("username") username: String): Observable<User>

    @GET("users")
    fun getAll(): Observable<List<User>>
}